iina.xcodeproj/project.pbxproj

```diff
@@ -4022,6 +4025,14 @@
 /* End XCConfigurationList section */
 
 /* Begin XCRemoteSwiftPackageReference section */
+		51D34D8A28DD49C1006C2C87 /* XCRemoteSwiftPackageReference "swift-atomics" */ = {
+			isa = XCRemoteSwiftPackageReference;
+			repositoryURL = "https://github.com/apple/swift-atomics.git";
```

Comment:
I don't think this deserves bringing in a new dependency. A Bool behind a Lock should be fine.

Reply:
Surprisingly the Swift Atomics [README](https://github.com/apple/swift-atomics) does not explain why you would want to use that package instead of just using mutexes. The reason is that using OS locks is several orders of magnitude slower than the atomic operations provided by Swift Atomics. That is due to this:

> all atomic operations map directly to dedicated CPU instructions where available -- to the extent supported by llvm & Clang.

This post from the ArangoDB site shows the high cost of OS locks vs. atomic operations: [Comparison: Lockless programming with atomics in C++ 11 vs. mutex and RW-locks](https://www.arangodb.com/2015/02/comparing-atomic-mutex-rwlocks/)

For this issue there is no need for optimal performance. But I was expecting atomics to be used to solve some of the various data races being reported by the ThreadSanitizer. Some of those involve operations that occur frequently. I've not dug into those issues yet. It may be that those could be solved by consistently queuing to a particular thread, rather than sharing between threads.

I can pull the use of Swift Atomics and change AtomicBoolean to use a lock. If we find a case were simple typed data is frequently accessed between two threads we could then decide if we should add use of Swift Atomics.

Does that sound good to you?

Reply:
I worked as an iOS performance engineer in the past, so I'm familiar with atomics and how they are used :) In this case I just don't think it is worth bringing in a dependency. When Swift adds these natively we can switch over to them, and until then we can always drop down to C if necessary to access these.

Reply:
Understood. Right now I'm thinking with the event driven approach and main thread dispatching access to the new flags can be restricted to the main thread, eliminating the need for this class in this case.

If it proves to be needed, I'll propose a version that uses the new `Lock` class and not Swift Atomics.


It would be good to get PR #3938 merged (logger improvements) as that one adds the `Lock` class.